flake.nix

```diff
@@ -0,0 +1,85 @@
+{
+  description = "yesod-static-remote";
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?ref=nixos-22.05";
+    flake-utils.url = "github:numtide/flake-utils";
+    pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
+  };
+
+  outputs =
+    { self
+    , nixpkgs
+    , flake-utils
+    , pre-commit-hooks
+    }:
+    flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+    let
+      pkgs = import nixpkgs {
+        inherit system;
+        overlays = [
+          self.overlays.${system}
+        ];
+      };
+    in
+    {
+      overlays = final: prev:
+        with final.haskell.lib;
+        {
+          haskellPackages =
+            prev.haskellPackages.override (
+              old:
+              {
+                overrides =
+                  final.lib.composeExtensions (old.overrides or (_: _: { })) (
+                    self: super: {
+                      yesod-static-remote = dontCheck (failOnAllWarnings (self.callPackage ./default.nix { }));
+                    }
+                  );
+              }
+            );
+        };
+
+      packages.yesod-static-remote = pkgs.haskellPackages.yesod-static-remote;
+      packages.default = self.packages.${system}.yesod-static-remote;
+      checks.pre-commit = pre-commit-hooks.lib.${system}.run {
+        src = ./.;
+        hooks = {
+          hlint.enable = true;
+          hpack.enable = true;
+          nixpkgs-fmt.enable = true;
+          ormolu.enable = true;
+          cabal2nix = {
+            enable = true;
+            name = "cabal2nix";
+            files = "\\.cabal$";
+            entry = "${pkgs.writeShellScript "cabal2nix-hook" ''
+              for cabalFile in $*
+              do
+                dir="$(dirname $cabalFile)"
+                defaultFile="$dir/default.nix"
+                ${pkgs.cabal2nix}/bin/cabal2nix "$cabalFile" > "$defaultFile"
+                ${pkgs.nixpkgs-fmt}/bin/nixpkgs-fmt "$defaultFile"
+              done
+
+            ''}";
+          };
+        };
+      };
+      devShells.default = pkgs.mkShell {
```

Comment:
Better to use `shellFor`? cf. https://github.com/kowainik/summoner/pull/335/files#diff-245563af9e34754739b12b294f3324c1d42c116ace7ce8735e21dfbc82d8e064R46